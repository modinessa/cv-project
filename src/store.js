import { configureStore } from "@reduxjs/toolkit";
import { navigationPanelReducer } from "./components/NavigationPanel/navigationPanelReducer";

export const store = configureStore({
	reducer: {
		navigation: navigationPanelReducer,
	}
});
