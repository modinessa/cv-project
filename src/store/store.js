import { configureStore } from "@reduxjs/toolkit";
import navigationReducer from "../components/NavigationPanel/navigationPanelReducer";

export const store = configureStore({
	reducer: {
		navigation: navigationReducer,
	}
});