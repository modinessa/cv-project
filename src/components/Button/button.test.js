import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { store } from '../../store/store';
import { Button } from './index';

const TEST_ICON = 'FontAwesomeIcon';
const TEST_TYPE_TRANS = 'link-transparent';
const TEST_TYPE_BTN = 'link-button';
const TEST_TYPE_SQR_T = 'square-top';
const TEST_TYPE_SQR_R = 'square-right';
const TEST_TYPE_SQR_B = 'square-bottom';
const TEST_TYPE_SQR_L = 'square-left';
const TEST_TYPE_SOME = 'some';
const TEST_TEXT = 'Texty bla-bla';

describe('Button', () => {
	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_BTN}
					cliclickHandler={mockCallBack}
					text={TEST_TEXT}
				/>
			</Provider>
		)).toMatchSnapshot();
	});

	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_TRANS}
					cliclickHandler={mockCallBack}
					text={TEST_TEXT}
				/>
			</Provider>
		)).toMatchSnapshot();
	});
	
	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_SQR_T}
					cliclickHandler={mockCallBack}
				/>
			</Provider>
		)).toMatchSnapshot();
	})
	
	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_SQR_R}
					cliclickHandler={mockCallBack}
				/>
			</Provider>
		)).toMatchSnapshot();
	})
	
	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_SQR_L}
					cliclickHandler={mockCallBack}
				/>
			</Provider>
		)).toMatchSnapshot();
	})
	
	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_SQR_B}
					cliclickHandler={mockCallBack}
				/>
			</Provider>
		)).toMatchSnapshot();
	})

	it('renders component', () => {
		const mockCallBack = jest.fn();
		expect(renderer.create(
			<Provider store={store}>
				<Button
					icon={TEST_ICON}
					type={TEST_TYPE_SOME}
					cliclickHandler={mockCallBack}
					text={TEST_TEXT}
				/>
			</Provider>
		)).toMatchSnapshot();
	})
})