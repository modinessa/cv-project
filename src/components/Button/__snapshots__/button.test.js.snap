// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component 1`] = `
<button
  className="app-button link-button"
>
  FontAwesomeIcon
   
  <span
    className="text"
  >
    Texty bla-bla
  </span>
</button>
`;

exports[`Button renders component 2`] = `
<button
  alt="Link"
  className="app-button link-transparent"
>
  FontAwesomeIcon
   
  <span
    className="text"
  >
    Texty bla-bla
  </span>
</button>
`;

exports[`Button renders component 3`] = `
<button
  className="app-button square-top"
>
  FontAwesomeIcon
</button>
`;

exports[`Button renders component 4`] = `
<button
  className="app-button square-right"
>
  FontAwesomeIcon
</button>
`;

exports[`Button renders component 5`] = `
<button
  className="app-button square-left"
>
  FontAwesomeIcon
</button>
`;

exports[`Button renders component 6`] = `
<button
  className="app-button square-bottom"
>
  FontAwesomeIcon
</button>
`;

exports[`Button renders component 7`] = `
<button
  className="app-button"
>
  FontAwesomeIcon
   
  <span
    className="text"
  >
    Texty bla-bla
  </span>
</button>
`;
